src/avatar/src/Avatar.js

```diff
@@ -21,167 +21,145 @@ const initialsStyleClass = css({
+
+  /**
+   * The name used for the initials and title attribute.
+   */
+  name: PropTypes.string,
+
+  /**
+   * The value used for the hash function.
+   * The name is used as the hashValue by default.
+   * When dealing with anonymous users you should use the id instead.
+   */
+  hashValue: PropTypes.string,
+
+  /**
+   * When true, render a solid avatar.
+   */
+  isSolid: PropTypes.bool,
+
+  /**
+   * The color used for the avatar.
+   * When the value is `automatic`, use the hash function to determine the color.
+   */
+  color: PropTypes.string.isRequired,
+
+  /**
+   * Function to get the initials based on the name.
+   */
+  getInitials: PropTypes.func,
+
+  /**
+   * When true, force show the initials.
+   * This is useful in some cases when using Gravatar and transparent pngs.
+   */
+  forceShowInitials: PropTypes.bool,
+
+  /**
+   * When the size is smaller than this number, use a single initial for the avatar.
+   */
+  sizeLimitOneCharacter: PropTypes.number
+}
 
-  render() {
-    const {
-      theme,
-
-      src,
-      size,
-      name,
-      isSolid,
-      hashValue: propsHashValue,
-      getInitials,
-      color: propsColor,
-      forceShowInitials,
-      sizeLimitOneCharacter,
-      ...props
-    } = this.props
-
-    const { imageHasFailedLoading } = this.state
-    const imageUnavailable = !src || imageHasFailedLoading
-    const initialsFontSize = `${theme.getAvatarInitialsFontSize(
-      size,
-      sizeLimitOneCharacter
-    )}px`
-
-    let initials = getInitials(name)
-    if (size <= sizeLimitOneCharacter) {
-      initials = initials.substring(0, 1)
-    }
-
-    const colorProps = this.getColorProps()
-
-    return (
-      <Box
-        width={size}
-        height={size}
-        overflow="hidden"
-        borderRadius={9999}
-        position="relative"
-        display="inline-flex"
-        flexShrink={0}
-        justifyContent="center"
-        backgroundColor={colorProps.backgroundColor}
-        title={name}
-        {...props}
-      >
-        {(imageUnavailable || forceShowInitials) && (
-          <Text
-            className={initialsStyleClass}
-            fontSize={initialsFontSize}
-            lineHeight={initialsFontSize}
-            width={size}
-            height={size}
-            color={colorProps.color}
-          >
-            {initials}
-          </Text>
-        )}
-        {!imageUnavailable && (
-          <Image
-            style={{ objectFit: 'cover' }} // Unsupported by ui-box directly
-            width={isObjectFitSupported ? '100%' : 'auto'} // Fallback to old behaviour on IE
-            height="100%"
-            src={src}
-            onError={this.handleError}
-          />
-        )}
-      </Box>
-    )
-  }
+Avatar.defaultProps = {
+  color: 'automatic',
+  size: 24,
+  isSolid: false,
+  getInitials: globalGetInitials,
+  forceShowInitials: false,
+  sizeLimitOneCharacter: 20
 }
 
-export default withTheme(Avatar)
+export default Avatar
```

Comment:
Same comment as in https://github.com/segmentio/evergreen/pull/724#discussion_r372533724 – for parity we should probably support `forwardRef` + `React.memo`?